src/sagemaker/serve/builder/model_builder.py

```diff
@@ -64,6 +64,7 @@
 )
 from sagemaker.workflow.entities import PipelineVariable
 from sagemaker.huggingface.llm_utils import get_huggingface_model_metadata
+from sagemaker_schema_inference_artifacts.huggingface import remote_schema_retriever
```

Comment:
I would suggest moving this import to be in the try catch block where we use remote_schema_retriever and catching an import error there if module is not installed.

The reason for the suggestion is: this dependencies are extra dependencies and not all the customers want to opt in to use this feature. example: https://github.com/aws/sagemaker-python-sdk/pull/4549